### PR TITLE
feat: remove the loca table order check

### PR DESCRIPTION
**Pull request by yisibl**
Some woff2 fonts `loca` table is not always behind the `glyf` table, such as the very popular Font Awesome(I added it to the test case). 

This PR removes those checks and adjusts to only check if the `loca` table exists.

The WOFF2 specification has further instructions:

- Spec: https://www.w3.org/TR/WOFF2/#table_order
  > The loca table MUST follow the glyf table in the table directory. When WOFF2 file contains individually encoded font file, the table directory MAY contain other tables inserted between glyf and loca tables; however when WOFF2 contains a font collection file each loca table MUST immediately follow its corresponding glyf table. For example, the following order of tables: 'cmap', 'glyf', 'hhea', 'hmtx', 'loca', 'maxp' ... is acceptable for individually encoded font files;

The file structure can be viewed via `woff2_info` in [google/woff2](https://github.com/google/woff2/blob/master/src/woff2_info.cc).

`woff2_info src/test_resources/lato-v22-latin-regular.woff2`

```bash
Entry offset flags tag  origLength txLength
    0     48  0x3f FFTM         28
    1     54  0x1a GDEF         30
    2     56  0x06 OS/2         96
    3     58  0x00 cmap        986
    4     61  0x11 gasp          8
    5     63  0x0a glyf      27996    21922
    6     70  0x01 head         54
    7     72  0x02 hhea         36
    8     74  0x03 hmtx        596
    9     77  0x0b loca        310        0
   10     81  0x04 maxp         32
   11     83  0x05 name       1371
   12     86  0x07 post       1963
```

### CODE REVIEW

**Comment by yisibl** [comment]
@zyxw59 PTAL

**Review by zyxw59 — COMMENTED** [review]
Is the `foo.ttf` file meant to be included?

**Comment by zyxw59** [comment]
I don't have write access to this repo anymore (as I no longer work at Cimpress), so I'll tag @BrianHanechak to take a look.

File: src/woff2/table_directory.rs [hunk 1]
```diff
@@ -126,8 +126,10 @@ impl Woff2TableDirectory {
                         );
                     }
                 }
-                // we handle `loca` table with `glyf` above
-                LOCA_TAG => return Err(WriteTablesError::MissingGlyfTable),
+                // - Spec: https://www.w3.org/TR/WOFF2/#table_order
+                // The loca table MUST follow the glyf table in the table directory. When WOFF2 file contains individually encoded font file, the table directory MAY contain other tables inserted between glyf and loca tables; For example, the following order of tables: 'cmap', 'glyf', 'hhea', 'hmtx', 'loca', 'maxp' ... is acceptable for individually encoded font files;
+                // TODO: however when WOFF2 contains a font collection file each loca table MUST immediately follow its corresponding glyf table.
+                LOCA_TAG => {}
```
[the remark this answers is not in the document]
Comment (yisibl) [reply src/woff2/table_directory.rs:131]: This can be implemented later.

**Comment by yisibl** [comment]
@zyxw59 Thank you very much for your review.

> Is the foo.ttf file meant to be included?

My bad, foo.ttf has been deleted.

**Comment by Shubhamj280** [comment]
Looks good to me! I will merge the MR.

**Pull request merged**